### PR TITLE
Fix Enum as string used with Has Issue

### PR DESCRIPTION
**Pull request by Sreejithpin**
### Issues
Fix Enum as string used with Has Issue
*This pull request fixes issue #xxx.*

### Description

*Briefly describe the changes of this pull request.*

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

**Comment by mikepizzo** [comment]
>                 "MoonSet/'Blue'",

proper syntax should be "MoonSet/Blue (without the quotes). Is this validated anywhere?

---
Refers to: test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Metadata/ODataUriResolverTests.cs:235 in 45735c8. [](commit_id = 45735c8f36d95f9d5bd975c0b04e2157b0e76561, deletion_comment = False)

**Comment by mikepizzo** [comment]
>                 "MoonSet2(id=1,color='Blue')",

does this succeed or return an error?

---
Refers to: test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Metadata/ODataUriResolverTests.cs:255 in 45735c8. [](commit_id = 45735c8f36d95f9d5bd975c0b04e2157b0e76561, deletion_comment = False)

**Comment by mikepizzo** [comment]
>                 "MoonSet('Blue')",

does this succeed, or is it expected to return an error?

---
Refers to: test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Metadata/ODataUriResolverTests.cs:218 in 45735c8. [](commit_id = 45735c8f36d95f9d5bd975c0b04e2157b0e76561, deletion_comment = False)

**Comment by mikepizzo** [comment]
Fix to the original issue looks good, but I'm curious whether the same issue happens with key values. are the tests no expecting unqualified enum values to be valid when used as key values?

**Comment by xuzhg** [comment]
>                 "MoonSet('Blue')",

I think it is correct, because it plays like a string, and a string should have single quote wrapped? 

---
In reply to: [681187915](https://github.com/OData/odata.net/pull/1865#issuecomment-681187915) [](ancestors = 681187915)

---
Refers to: test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Metadata/ODataUriResolverTests.cs:218 in 45735c8. [](commit_id = 45735c8f36d95f9d5bd975c0b04e2157b0e76561, deletion_comment = False)

File: test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Metadata/ODataUriResolverTests.cs [hunk 1]
```diff
@@ -78,6 +78,14 @@ private static void VerifyEnumVsStringFilterExpression(FilterClause filter)
             bin.Right.ShouldBeEnumNode(enumtypeRef.EnumDefinition(), "2");
         }
 
+        private static void VerifyEnumVsStringFilterExpressionforHas(FilterClause filter)
```
Comment (xuzhg) [inline test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Metadata/ODataUriResolverTests.cs:81]: If this method is only used at one place, maybe remove this function and add the codes into the test directly?

**Comment by Sreejithpin** [comment]
> See my comments. Do we need a separate issue to track use of enum as string in keys?

Lets discuss in the call

**Pull request merged**